crates/sui-framework/packages/bridge/tests/bridge_tests.move

```diff
@@ -0,0 +1,693 @@
+            0, // freeze op
+        );
+        let payload = msg.extract_emergency_op_payload();
+        inner.test_execute_emergency_op(payload);
+        inner.assert_paused(error);
+    }
+
+    // unfreeze the bridge
+    fun unfreeze_bridge(bridge: &mut Bridge, error: u64) {
+        let inner = bridge.test_load_inner_mut();
+        // unfreeze it
+        let msg = message::create_emergency_op_message(
+            chain_ids::sui_testnet(),
+            1, // seq num, this is supposed to be the next seq num but it's not what we test here
+            1, // unfreeze op
+        );
+        let payload = msg.extract_emergency_op_payload();
+        inner.test_execute_emergency_op(payload);
+        inner.assert_not_paused(error);
+    }
+
+    #[test]
+    fun test_bridge_create() {
+        let mut scenario = test_scenario::begin(@0x0);
+        create_bridge_default(&mut scenario);
+
+        scenario.next_tx(@0xAAAA);
+        let mut bridge = scenario.take_shared<Bridge>();
+        let inner = bridge.test_load_inner_mut();
+        inner.assert_not_paused(UNEXPECTED_ERROR);
+        assert!(inner.inner_token_transfer_records().length() == 0, UNEXPECTED_ERROR + 1);
+
+        test_scenario::return_shared(bridge);
+        scenario.end();
+    }
+
+    #[test]
+    #[expected_failure(abort_code = bridge::bridge::ENotSystemAddress)]
+    fun test_bridge_create_non_system_addr() {
+        let mut scenario = test_scenario::begin(@0x1);
+        create_bridge(&mut scenario, @0x1);
+
+        abort TEST_DONE
+    }
+
+    #[test]
+    fun test_init_committee() {
+        let mut scenario = test_scenario::begin(@0x0);
+
+        create_bridge_default(&mut scenario);
+        register_committee(&mut scenario);
+        init_committee(&mut scenario, @0x0);
+
+        scenario.end();
+    }
+
+    #[test]
+    fun test_init_committee_twice() {
+        let mut scenario = test_scenario::begin(@0x0);
+
+        create_bridge_default(&mut scenario);
+        register_committee(&mut scenario);
+        init_committee(&mut scenario, @0x0);
+        init_committee(&mut scenario, @0x0); // second time is a no-op
+
+        scenario.end();
+    }
+
+    #[test]
+    #[expected_failure(abort_code = bridge::bridge::ENotSystemAddress)]
+    fun test_init_committee_non_system_addr() {
+        let mut scenario = test_scenario::begin(@0x0);
+
+        create_bridge_default(&mut scenario);
+        register_committee(&mut scenario);
+        init_committee(&mut scenario, @0xA);
+
+
+        abort TEST_DONE
+    }
+
+    #[test]
+    #[expected_failure(abort_code = bridge::committee::ECommitteeAlreadyInitiated)]
+    fun test_register_committee_after_init() {
+        let mut scenario = test_scenario::begin(@0x0);
+
+        create_bridge_default(&mut scenario);
+        register_committee(&mut scenario);
+        init_committee(&mut scenario, @0x0);
+        register_committee(&mut scenario);
+
+
+        abort TEST_DONE
+    }
+
+    // #[test]
```

Comment:
why do we have these commented?

Reply:
because they do not work yet as I need to build more infra for those tests... I did not want to have them completely out yet, which I could if we are going to get this in as it is